File: server/src/main/java/com/shoptourr/api/v1/dto/push/PushDtos.java
package com.shoptourr.api.v1.dto.push;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import java.time.Instant;
import java.util.UUID;

/**
 * Push device registration — budget alerts (P2).
 */
public final class PushDtos {

    private PushDtos() {}

    public enum PushPlatform {
        ANDROID,
        IOS
    }

    public record RegisterDeviceRequest(
            @NotBlank @Size(max = 512) String token,
            @NotNull PushPlatform platform,
            @Size(max = 64) String appVersion,
            @Size(max = 120) String deviceName
    ) {}

    public record DeviceDto(
            UUID id,
            String tokenFingerprint,
            PushPlatform platform,
            String appVersion,
            String deviceName,
            Instant createdAt,
            Instant lastSeenAt
    ) {}
}

File: shared/src/commonMain/kotlin/com/example/shoptourr/data/remote/dto/push/PushDtos.kt
package com.example.shoptourr.data.remote.dto.push

import kotlinx.serialization.Serializable

@Serializable
enum class PushPlatformDto {
    ANDROID,
    IOS,
}

@Serializable
data class RegisterDeviceRequest(
    val token: String,
    val platform: PushPlatformDto,
    val appVersion: String? = null,
    val deviceName: String? = null,
)

@Serializable
data class DeviceDto(
    val id: String,
    val tokenFingerprint: String,
    val platform: PushPlatformDto,
    val appVersion: String? = null,
    val deviceName: String? = null,
    val createdAt: String,
    val lastSeenAt: String? = null,
)
